nullaway/src/test/java/com/uber/nullaway/tools/SerializationTestHelper.java

```diff
@@ -157,4 +159,47 @@ private List<T> readActualOutputs(Path outputPath) {
     }
     return outputs;
   }
+
+  /**
+   * Checks if given paths are equal. Under different OS environments, identical paths might have a
+   * different string representation. In windows all forward slashes are replaced with backslashes.
+   *
+   * @param expected Expected serialized path.
+   * @param found Serialized path.
+   * @return true, if paths are identical.
+   */
+  public static boolean pathsAreEqual(String expected, String found) {
+    if (found.equals(expected)) {
+      return true;
+    }
+    return found.replaceAll("\\\\", "/").equals(expected);
```

Comment:
Is it really possible to hit this case and have it be `true` when dealing with paths that were serialized and deserialized in the same system? Or are we dealing with comparisons of URIs converted directly to Strings vs Paths directly converted to String? What test fails if this is replaced by string equality?

p.s. Also, why are we comparing these as strings instead of deserializing them as `Path`? Because we need to support `"null"`?

Reply:
Actually, never mind, I get why we need this `.replaceAll(...)` call. But the proper fix I think would be e.g. [here](https://github.com/uber/NullAway/blob/master/nullaway/src/test/java/com/uber/nullaway/NullAwaySerializationTest.java#L258), instead of passing the system specific string representation `"com/uber/android/Test.java"`, these should be `Paths.get("com", "uber", "android", "Test.java")` and so on... but that might be a lot of changes of existing test cases right now, so definitely worth leaving for a follow up.